File: CS/WindowsFormsApp3/Form1.cs
using DevExpress.DashboardCommon;
using System.Windows.Forms;

namespace WindowsFormsApp3 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            dashboardDesigner1.AsyncDataLoading += (sender, e) => {
                e.Data = DataRow.GetData();
            };

            Dashboard dashboard = new Dashboard();
            DashboardObjectDataSource ds = new DashboardObjectDataSource(new object());
            dashboard.DataSources.Add(ds);
            GridDashboardItem grid = new GridDashboardItem() { DataSource = ds };
            grid.Columns.Add(new GridDimensionColumn(new Dimension("Dimension1")));
            grid.Columns.Add(new GridDimensionColumn(new Dimension("Dimension2")));
            grid.Columns.Add(new GridDimensionColumn(new Dimension("Dimension3")));
            grid.Columns.Add(new GridMeasureColumn(new Measure("Measure")));
            grid.InteractivityOptions.IsDrillDownEnabled = true;
            dashboard.Items.Add(grid);
            ChartDashboardItem chart = new ChartDashboardItem() { DataSource = ds };
            chart.Arguments.Add(new Dimension("Dimension1"));
            chart.Arguments.Add(new Dimension("Dimension2"));
            chart.Arguments.Add(new Dimension("Dimension3"));
            ChartPane pane = new ChartPane();
            pane.Series.Add(new SimpleSeries(new Measure("Measure")));
            chart.Panes.Add(pane);
            chart.InteractivityOptions.IsDrillDownEnabled = true;
            dashboard.Items.Add(chart);

            dashboardDesigner1.Dashboard = dashboard;

            new MasterDrillDownModule(dashboardDesigner1);
        }
    }
}

File: VB/WinForms_Dashboard_Drill-Down/Form1.vb
Imports System
Imports DevExpress.DashboardCommon
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WinForms_Dashboard_Drill_Down
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			dashboardDesigner1.CreateRibbon()
			dashboardDesigner1.AsyncMode = True
			AddHandler dashboardDesigner1.AsyncDataLoading, Sub(sender, e)
				e.Data = DataRow.GetData()
			End Sub
			Dim dashboard As New Dashboard()
			Dim ds As New DashboardObjectDataSource(New Object())
			dashboard.DataSources.Add(ds)
			Dim grid As New GridDashboardItem() With {.DataSource = ds}
			grid.Columns.Add(New GridDimensionColumn(New Dimension("Dimension1")))
			grid.Columns.Add(New GridDimensionColumn(New Dimension("Dimension2")))
			grid.Columns.Add(New GridDimensionColumn(New Dimension("Dimension3")))
			grid.Columns.Add(New GridMeasureColumn(New Measure("Measure")))
			grid.InteractivityOptions.IsDrillDownEnabled = True
			dashboard.Items.Add(grid)
			Dim chart As New ChartDashboardItem() With {.DataSource = ds}
			chart.Arguments.Add(New Dimension("Dimension1"))
			chart.Arguments.Add(New Dimension("Dimension2"))
			chart.Arguments.Add(New Dimension("Dimension3"))
			Dim pane As New ChartPane()
			pane.Series.Add(New SimpleSeries(New Measure("Measure")))
			chart.Panes.Add(pane)
			chart.InteractivityOptions.IsDrillDownEnabled = True
			dashboard.Items.Add(chart)

			dashboardDesigner1.Dashboard = dashboard

			Dim tempVar As New MasterDrillDownModule(dashboardDesigner1)
		End Sub
	End Class
End Namespace
